source/views/dictionary/report/editor.tsx

```diff
@@ -10,80 +10,66 @@ import {
 import {submitReport} from './submit'
 import type {WordType} from '../types'
 import type {TopLevelViewPropsType} from '../../types'
+import {NativeStackNavigationOptions} from '@react-navigation/native-stack'
+import noop from 'lodash/noop'
 
 type Props = TopLevelViewPropsType & {
-	navigation: {state: {params: {item: WordType}}}
+	route: {params: {item: WordType}}
 }
 
-type State = {
-	term: string
-	definition: string
+export const NavigationOptions: NativeStackNavigationOptions = {
+	title: 'Suggest an edit',
+	headerBackTitle: 'Back',
 }
 
-export class DictionaryEditorView extends React.PureComponent<Props, State> {
-	static navigationOptions = () => {
-		return {
-			title: 'Suggest an Edit',
-		}
-	}
-
-	state = {
-		term: this.props.navigation.state.params.word.word,
-		definition: this.props.navigation.state.params.word.definition,
-	}
+let DictionaryEditorView = (props: Props): JSX.Element => {
+	let [term, setTerm] = React.useState(props.route.params.item.word)
+	let [definition, setDefinition] = React.useState(
+		props.route.params.item.definition,
+	)
 
-	submit = () => {
-		submitReport(this.props.navigation.state.params.word, {
-			word: this.state.term.trim(),
-			definition: this.state.definition.trim(),
+	let submit = () => {
+		submitReport(props.route.params.item, {
+			word: term.trim(),
+			definition: definition.trim(),
 		})
 	}
 
-	onChangeTitle = (newTitle: string) => {
-		this.setState(() => ({term: newTitle}))
-	}
-
-	onChangeDefinition = (newDefinition: string) => {
-		this.setState(() => ({definition: newDefinition}))
-	}
-
-	render() {
-		let {term, definition} = this.state
-
-		return (
-			<ScrollView
-				keyboardDismissMode="on-drag"
-				keyboardShouldPersistTaps="always"
-			>
-				<InfoHeader
-					message="If you could tell us what the word and definition should be, we&rsquo;d greatly appreciate it."
-					title="Thanks for spotting a problem!"
-				/>
+	return (
+		<ScrollView
+			keyboardDismissMode="on-drag"
+			keyboardShouldPersistTaps="always"
+		>
+			<InfoHeader
+				message="If you could tell us what the word and definition should be, we&rsquo;d greatly appreciate it."
+				title="Thanks for spotting a problem!"
+			/>
 
-				<TableView>
-					<Section header="WORD">
-						<TitleCell onChange={this.onChangeTitle} text={term} />
-					</Section>
+			<TableView>
+				<Section header="WORD">
+					<TitleCell onChange={(newTitle) => setTerm(newTitle)} text={term} />
+				</Section>
 
-					<Section header="DEFINITION">
-						<DefinitionCell
-							onChange={this.onChangeDefinition}
-							text={definition}
-						/>
-					</Section>
+				<Section header="DEFINITION">
+					<DefinitionCell
+						onChange={(newDefinition) => setDefinition(newDefinition)}
```

Comment:
How does this read? Too confusing?

```suggestion
						onChange={setDefinition}
```

Reply:
I did consider that syntax to inline that. What do you think, @rye?

Reply:
For singleton `handler={(arg) => someFunction(arg)}` situations like this, I definitely think the `handler={someFunction}` is a good way to go, especially since that's a `useState` delegate, which isn't likely to sprout a new parameter any time soon.